addon/-private/system/store/finders.js

```diff
@@ -202,12 +202,15 @@ function ensureRelationshipIsSetToParent(
         until: '3.8',
       });
     }
-    relationships[inverseKey] = relationships[inverseKey] || {};
-    relationships[inverseKey].data = fixRelationshipData(
-      relationshipData,
-      kind,
-      parentInternalModel
-    );
+
+    if (typeof relationshipData !== 'undefined') {
```

Comment:
I suspect we can be a little bit more careful here by checking if it's a `hasMany` relationship, as a `belongsTo` would never exhibit this bug behavior and should be fixed.

Reply:
Changed it to only check for undefined on hasMany relationships